samples/client/petstore/kotlin-gson/src/main/kotlin/org/openapitools/client/apis/PetApi.kt

```diff
@@ -75,6 +72,24 @@ class PetApi(basePath: kotlin.String = defaultBasePath) : ApiClient(basePath) {
         }
     }
 
+    /**
+    * Add a new pet to the store
+    * 
+    * @param body Pet object that needs to be added to the store 
+    * @return ApiInfrastructureResponse<Unit?>
+    * @throws UnsupportedOperationException If the API returns an informational or redirection response
+    * @throws ClientException If the API returns a client error response
+    * @throws ServerException If the API returns a server error response
+    */
+    @Throws(UnsupportedOperationException::class, ClientException::class, ServerException::class)
+    fun addPetWithHttpInfo(body: Pet) : ApiInfrastructureResponse<Unit?> {
+        val localVariableConfig = addPetRequestConfig(body = body)
+
+        return request<Pet, Unit>(
```

Comment:
I'm not sure if here we should return the ApiInfrastructureResponse or if we should return only the success response, and throw the other errors, something like:
```
    @Suppress("UNCHECKED_CAST")
    @Throws(UnsupportedOperationException::class, ClientException::class, ServerException::class)
    fun findPetsByStatus(status: kotlin.collections.List<kotlin.String>) : kotlin.collections.List<Pet> {
        val localVarResponse = findPetsByStatusWithHttpInfo(status = status)

        return localVarResponse.data
    }

    @Suppress("UNCHECKED_CAST")
    @Throws(UnsupportedOperationException::class, ClientException::class, ServerException::class)
    fun findPetsByStatusWithHttpInfo(status: kotlin.collections.List<kotlin.String>) : Success<kotlin.collections.List<Pet>> {
        val localVariableConfig = findPetsByStatusRequestConfig(status = status)

        val localVarResponse = request<Unit, kotlin.collections.List<Pet>>(
            localVariableConfig
        )

        return when (localVarResponse.responseType) {
            ResponseType.Success -> localVarResponse
            ResponseType.Informational -> throw UnsupportedOperationException("Client does not support Informational responses.")
            ResponseType.Redirection -> throw UnsupportedOperationException("Client does not support Redirection responses.")
            ResponseType.ClientError -> {
                val localVarError = localVarResponse as ClientError<*>
                throw ClientException("Client error : ${localVarError.statusCode} ${localVarError.message.orEmpty()}", localVarError.statusCode, localVarResponse)
            }
            ResponseType.ServerError -> {
                val localVarError = localVarResponse as ServerError<*>
                throw ServerException("Server error : ${localVarError.statusCode} ${localVarError.message.orEmpty()}", localVarError.statusCode, localVarResponse)
            }
        }
    }
```

Reply:
I guess that works for my use case, but why not enable custom use cases by returning everything? In general I favour passing information unchanged if possible

Reply:
@wing328 what do you think about this? Should we just return the raw response or do some king of error handling?

Reply:
Maybe we should just keep the @wegendt-bosch implementation of returning the raw response because that's the more flexible approach?